### PR TITLE
[rush] Surface dependencySettings on the deploy.json configuration

### PR DESCRIPTION
**Pull request by EscapeB**
## Summary
Add dependencySettings field to Rush deploy.json configurations. This will allow developers to customize how third party dependencies are processed when running `rush deploy` 
Related PR https://github.com/microsoft/rushstack/pull/4255 cc @D4N14L 

## How it was tested
Tested locally

## Impacted documentation
Schema for deploy.json has been updated and need to be updated.

### CODE REVIEW

File: libraries/rush-lib/src/schemas/deploy-scenario.schema.json [hunk 1]
```diff
@@ -94,6 +94,40 @@
...
+      "type": "array",
+      "items": {
+        "type": "object",
+        "properties": {
+          "dependencyName": {
+            "description": "The full package name of third party dependency",
+            "type": "string"
+          },
+          "dependencyVersionRange": {
+            "description": "The semver version range of third party dependency",
```
Comment (dmichon-msft) [inline libraries/rush-lib/src/schemas/deploy-scenario.schema.json:110]: Nit: either "semantic version range" or "semver range".

**Review by dmichon-msft — APPROVED** [review]
Looks good, just a nit on the field description.

File: libraries/rush-lib/src/schemas/deploy-scenario.schema.json (hunk 1 above)
Comment (EscapeB) [reply libraries/rush-lib/src/schemas/deploy-scenario.schema.json]: Fixed~

File: libraries/rush-lib/src/schemas/deploy-scenario.schema.json [hunk 2]
```diff
@@ -94,6 +94,40 @@
...
+          "dependencyName": {
+            "description": "The full package name of third party dependency",
+            "type": "string"
+          },
+          "dependencyVersionRange": {
+            "description": "The semantic version range of third party dependency",
+            "type": "string"
+          },
+          "patternsToInclude": {
+            "description": "A list of glob patterns to include when extracting specific dependency. If a path is matched by both \"patternsToInclude\" and \"patternsToExclude\", the path will be excluded. If undefined, all paths will be included.",
```
Comment (iclanton) [inline libraries/rush-lib/src/schemas/deploy-scenario.schema.json:114]:
```suggestion
            "description": "A list of glob patterns to include when extracting the dependency specified in this object. If a path is matched by both \"patternsToInclude\" and \"patternsToExclude\", the path will be excluded. If undefined, all paths will be included.",
```

File: libraries/rush-lib/src/schemas/deploy-scenario.schema.json [hunk 3]
```diff
@@ -94,6 +94,40 @@
...
+            "type": "array",
+            "items": {
+              "type": "string"
+            }
+          },
+          "patternsToExclude": {
+            "description": "A list of glob patterns to exclude when extracting specific dependency. If a path is matched by both \"patternsToInclude\" and \"patternsToExclude\", the path will be excluded. If undefined, no paths will be excluded.",
```
Comment (iclanton) [inline libraries/rush-lib/src/schemas/deploy-scenario.schema.json:121]:
```suggestion
            "description": "A list of glob patterns to include when extracting the dependency specified in this object. If a path is matched by both \"patternsToInclude\" and \"patternsToExclude\", the path will be excluded. If undefined, no paths will be excluded.",
```

**Pull request merged**